perllib/FixMyStreet/Cobrand/Bromley.pm

```diff
@@ -1146,6 +1153,255 @@ sub garden_waste_cost {
     return $self->feature('payment_gateway')->{ggw_cost} * $bin_count;
 }
 
+sub waste_payment_type {
+    my ($self, $type, $ref) = @_;
+
+    my ($sub_type, $category);
+    if ( $type eq 'Payment: 01' ) {
+        $category = 'Garden Subscription';
+        $sub_type = $self->waste_subscription_types->{New};
+    } elsif ( $type eq 'Payment: 17' ) {
+        $category = 'Garden Subscription';
+        if ( $ref ) {
+            $sub_type = $self->waste_subscription_types->{Amend};
+        } else {
+            $sub_type = $self->waste_subscription_types->{Renew};
+        }
+    } elsif ( $type eq 'AUDDIS: 0C' ) {
+        $sub_type = 0;
+        $category = 'Cancel';
+    }
+
+    return ($category, $sub_type);
+}
+
+sub waste_reconcile_direct_debits {
+    my $self = shift;
+
+    my $today = DateTime->now;
+    my $start = $today->clone->add( days => -4 );
+
+    my $config = $self->feature('payment_gateway');
+    my $i = Integrations::Pay360->new({
+        config => $config
+    });
+
+    my $recent = $i->get_recent_payments(
+        start => $start,
+        end => $today
+    );
+
+
+    RECORD: for my $payment ( @$recent ) {
+
+        my $date = $payment->{CollectionDate};
+        my ($category, $type) = $self->waste_payment_type ( $payment->{Type}, $payment->{YourRef} );
+
+        next unless $category && $date;
+
+        my $payer = $payment->{PayerReference};
+        my $reference = $payer;
+
+        (my $uprn = $payer) =~ s/^GGW//;
+
+        my $len = length($uprn);
+        my $rs = FixMyStreet::DB->resultset('Problem')->search({
+            extra => { like => '%uprn,T5:value,I' . $len . ':'. $uprn . '%' },
+        },
+        {
+                order_by => { -desc => 'created' }
+        });
+
+        my $handled;
+
+        if ( $type && $type eq $self->waste_subscription_types->{Renew} ) {
+            next unless $payment->{Status} eq 'Paid';
+            $rs = $rs->search({ category => 'Garden Subscription' });
+            my $p;
+            while ( my $cur = $rs->next ) {
+                my $sub_type = $cur->get_extra_field_value('Subscription_Type');
+                if ( $sub_type eq $self->waste_subscription_types->{New} ) {
+                    $p = $cur;
+                } elsif ( $sub_type eq $self->waste_subscription_types->{Renew} ) {
+                    # already processed
+                    next RECORD if $cur->get_extra_metadata('dd_date') && $cur->get_extra_metadata('dd_date') eq $date;
+                }
+            }
+            if ( $p ) {
+                my $service = $self->waste_get_current_garden_sub( $p->get_extra_field_value('property_id') );
```

Comment:
Is property_id always going to be same as $uprn?

Reply:
I don't understand this comment.

Reply:
You've set $uprn a few lines up, but then are fetching this property_id here - if it's always going to be the same as $uprn, you could use that? Oh, as I have written that I realise that it is the ID not the UPRN, so can ignore me.